### ISSUE TITLE
tf Normalization 

### ISSUE THREAD

**Issue by theraghavjuneja** [tensorflow/tensorflow#64814]
### Issue type

Support

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.16.1

### Custom code

Yes

### OS platform and distribution

Windows

### Python version

3.11

### Current behavior?

Referring to tf.keras.layers.Normalization

In the documentation, I read that adapt-data is used so that tensor-flow can use the adapt data for computing means and variance corresponding to each axes. And axes can be specified as None if you want to compute the overall mean and variance of adapt data. It was mentioned either u can use the adapt data or explicitly specify mean and variance . Explicitly specifiying mean and variance doesnt work until or unless data is of the format np.array([[6.], [7.], [3.],[4.],[5.]] But if it is of np.array([[6,7,3,4,5]])/ A little difference in shape, This explicit will not work
Although It used to work in the back.

### Standalone code to reproduce the issue

```shell
normalizer=keras.layers.Normalization(mean=5,variance=4,axis=-1)
x_normalized=tf.constant([3,4,5,6,7],dtype='float32')
normalizer.adapt(x_normalized)
normalizer(x_normalized)
Whether axis is -1 or None this code won't work

But the below one would do 
input_data=tf.constant([[3.],[4.],[5.],[6.],[7.]],dtype='float32')
layer = keras.layers.Normalization(mean=5., variance=4.)
layer(input_data)
Now, another thing if I specify the adapt layer
adapt_data=np.array([[1.,2.,3.,4.,5.],
                     [3.,4.,5.,6.,7.],
                     [4.,5.,6.,7.,8.]],dtype='float32')
input_data=np.array([[3.,4.,5.,6.,7.]],dtype='float32')
layer = keras.layers.Normalization(axis=-1)
layer.adapt(adapt_data)
layer(input_data)
The above code of adapt layer will work but the same if I try to specify the same mean and variance corr to each axes or as an whole overall, it won't 
Although, this shoudln't be the case since explici specifying is allowed  here

EVEN AFTER USING RESHAPE , IT WON'T WORK
```

### Relevant log output

```shell
{{function_node __wrapped_}}
```

**Comment by theraghavjuneja** [comment]
Oh sorry! That was my fault
